drv/stm32h7-sprot-server/src/main.rs

```diff
@@ -401,61 +358,103 @@ impl ServerImpl {
 
                 // Intact messages from the RoT may indicate an error on
                 // its side.
-                Ok(rxmsg) => {
-                    match rxmsg.0.msgtype {
+                Ok(header) => {
+                    match header.msgtype {
                         MsgType::ErrorRsp => {
-                            if rxmsg.0.payload_len != 1 {
+                            if header.payload_len != 1 {
                                 ringbuf_entry!(Trace::ErrRspPayloadSize(
-                                    rxmsg.0.payload_len
+                                    header.payload_len
                                 ));
                                 // Treat this as a recoverable error
+                                hl::sleep_for(RETRY_TIMEOUT);
+                                ringbuf_entry!(Trace::ErrRespNoPayload);
                                 continue;
                             }
-                            let payload = &self.rx_buf.payload(&rxmsg);
-                            errcode = SprotError::from(payload[0]);
+                            errcode =
+                                SprotError::from(rxmsg.payload_error_byte());
                             ringbuf_entry!(Trace::SprotError(errcode));
-                            if matches!(
-                                errcode,
-                                SprotError::FlowError
-                                    | SprotError::InvalidCrc
-                                    | SprotError::UnsupportedProtocol
-                            ) {
+                            if is_recoverable_error(errcode) {
                                 // TODO: There are rare cases where
                                 // the RoT dose not receive
                                 // a 0x01 as the first byte in a message.
                                 // See issue #929.
+                                hl::sleep_for(RETRY_TIMEOUT);
+                                ringbuf_entry!(Trace::Recoverable(errcode));
                                 continue;
                             }
                             // Other errors from RoT are not recoverable with
                             // a retry.
                             return Err(errcode);
                         }
                         // All of the non-error message types are ok here.
-                        _ => return Ok(rxmsg),
+                        _ => return Ok(header),
                     }
                 }
             }
         }
         Err(errcode)
     }
 
+    // TODO: Move README.md to RFD 317 and discuss:
+    //   - Unsolicited messages from RoT to SP.
+    //   - Ignoring message from RoT to SP.
+    //   - Should we send a message telling RoT that SP has booted?
+    //
+    // The majority of this is documented in comments in the
```

Comment:
This part of the comment is out of date. Remove and replace with prior comment section:
```
   // For now, we are surprised that ROT_IRQ is asserted
   // But it would be ok to overlap our new request with receiving
   // of a previous response.
 ```